### DISCUSSION TITLE
up-validate=form targetting the wrong form

### DISCUSSION THREAD

**Discussion by mschoettle** [unpoly/unpoly#474] [Q&A]
Hello,

I am trying to adopt unpoly in a project that uses Django on the backend. The page contains a navigation menu with a dropdown that contains a link to sign out (it is based on this example here: https://getbootstrap.com/docs/5.3/examples/sidebars/).

Due to a recent change in Django, the logout needs to be done via a POST request instead of a GET. I did that via a `form` and a `button` inside it.

I now have a form in the main area of the page with dynamic fields. I added `up-validate="form"` to one of the form fields (a `<select>`) and when selecting an element the content disappears and the Log Out button appears.

According to the documentation for `up-validate` I understood it that it selects the closest form. Is it because `"form"` is a selector and selects the first form in the HTML? As I wrote this post I noticed that just `up-validate` works (probably because an empty target selects `:main`).

Is there a way to target the parent or closest form?

**Comment by adam12** [comment]
At least on v2 of Unpoly, this scenario is often because Unpoly tries to determine the selector to use and it might guess wrong. On v3 it's improved and should complain.

Can you confirm which version of Unpoly you're using? And I'm not sure if there'd be anything valuable in it, but try enabling the Unpoly logs and report back what it shows. `up.log.enable()` from your browser console.

You could try helping Unpoly by narrowing down a better selector, such as `#the-form-id` if you want to update the entire `<form>`, or look into [`[up-form-group]`](https://unpoly.com/up-form-group) depending on your Unpoly version (I _think_ it's v3 only).

**Reply by mschoettle** [reply]
Thanks for your reply. True, I forgot to mention that it is v3.

I guess `up-validate="form"` will select the first form in the HTML (which is the wrong one in my case). Basically was wondering whether that this is as intended.

I did change it to `up-validate=""` which then works as expected. I can see in the request that then targets the fieldset which contains this field.

**Reply by triskweline** [reply]
> I guess up-validate="form" will select the first form in the HTML (which is the wrong one in my case). Basically was wondering whether that this is as intended.

Does it update a different form on the current page?
Or does it update the expected form, but uses the wrong content from a response containing multiple forms?

**Reply by mschoettle** [reply]
Sorry for the delay. What I remember is that it POST'ed to the wrong form and replaced the response on the expected form (so the wrong content then).

I just tried to reproduce this with unpoly v3.2.0 and was unable to do so.
